### DISCUSSION TITLE
Request: long lasting URLs for Algerian channels

### DISCUSSION THREAD

**Discussion by ARAB-IPTV** [iptv-org/community#100] [General]
if you can make a long time url for algerians channels from the site htatv.com
like this :
[[http://free.fullspeed.tv/iptv-query?streaming-ip=https://www.htatv.com/channel/6

**Comment by BellezaEmporium** [comment]
Not possible, as said before, HTA's wmsAuthSign are IP-locked, thus, anything done from the Fullspeed URL will not work with the end user, since IPs are different.

**Comment by BellezaEmporium** [comment]
http://prox.x86.co.uk/index.php?https://cdn02.hta.dz/abr_htatv/PROGRAMME_NATIONAL/playlist.m3u8?wmsAuthSign=c2VydmVyX3RpbWU9MDMvMDUvMjAyMiAxMjo1NzozNSBQTSZoYXNoX3ZhbHVlPXhaUU1vbDQ0TG15QTZSREpmUXFJNkE9PSZ2YWxpZG1pbnV0ZXM9MTA=

Try this with a Firefox user-agent. One issue is that the HTA links are sort of "malformed", meaning that it doesn't keep the URL structure when getting links. We shall need to keep the URL as an "indicator" and add it over and over again to each part.

**Reply by BellezaEmporium** [reply]
@dct-infra Is there any chance that there's enough bandwidth or space for a miniProxy inside your infrastructure ? I think I found how to solve the IP-locked things.

**Reply by dct-infra** [reply]
@LaneSh4d0w I might have some resources available.
Do you have a repo for it?

**Reply by BellezaEmporium** [reply]
> @LaneSh4d0w I might have some resources available.
> Do you have a repo for it?

Haven't coded it so far, i'll keep you updated.

**Reply by BellezaEmporium** [reply]
> > @LaneSh4d0w I might have some resources available.
> > Do you have a repo for it?
> 
> Haven't coded it so far, i'll keep you updated.

MiniProxy is here : https://github.com/seopanel/QuickWebProxy/blob/master/miniProxy.php

I'll need to fix the m3u answers, get the token and it shall work.

**Reply by dct-infra** [reply]
@LaneSh4d0w Sorry but can't host PHP.

**Reply by BellezaEmporium** [reply]
> @LaneSh4d0w Sorry but can't host PHP.

There's this too : https://www.npmjs.com/package/miniproxy

**Reply by ARAB-IPTV** [reply]
there is another site that shows some algerians channels .
[iptv.at.dz](url)
when i extract a link in .mpd shown black screen .
can you work on it ...!

**Reply by BellezaEmporium** [reply]
> there is another site that shows some algerians channels .
> [iptv.at.dz](url)
> when i extract a link in .mpd shown black screen .
> can you work on it ...!

Those simply do not work for me.

**Comment by ARAB-IPTV** [comment]
hello friends *
this is a l nk of canal algerie tv
http://www.giniko.com/watch.php?id=328
i can extract m3u8 link but its dont stay a long time 
https://gintna100.tulix.tv/gin_live/canalalgerie/e4e0e16f6014c5bd33231ce972af18d3/6236fdc2/index.m3u8
-
i found this link but i this its work only on kodi , so can you create a long time url for this channel 

#EXTINF:-1 tvg-logo="http://www.giniko.com/logos/190x110/328.png" group-title="Giniko USA",DZ | Canal Algerie (SLT)
plugin://plugin.video.streamlink-tester/?action=play&url=http://www.giniko.com/watch.php?id=328

**Reply by BellezaEmporium** [reply]
> ## hello friends *
> this is a l nk of canal algerie tv
> http://www.giniko.com/watch.php?id=328
> i can extract m3u8 link but its dont stay a long time
> https://gintna100.tulix.tv/gin_live/canalalgerie/e4e0e16f6014c5bd33231ce972af18d3/6236fdc2/index.m3u8
> 
> i found this link but i this its work only on kodi , so can you create a long time url for this channel
> 
> #EXTINF:-1 tvg-logo="http://www.giniko.com/logos/190x110/328.png" group-title="Giniko USA",DZ | Canal Algerie (SLT) plugin://plugin.video.streamlink-tester/?action=play&url=http://www.giniko.com/watch.php?id=328

I have this personally http://nimlive6.giniko.com/canalalgerie/canalalgerie/playlist.m3u8?wmsAuthSign=c2VydmVyX3RpbWU9My8xOS8yMDIyIDc6MjA6NTEgUE0maGFzaF92YWx1ZT1uNnlteFRGZFcyMDQxN21YcExYVmJ3PT0mdmFsaWRtaW51dGVzPTE0NDA=

but since Giniko has a good amount of channels, i'll try to do a streamlink plugin.

**Reply by BellezaEmporium** [reply]
@dct-infra https://github.com/LaneSh4d0w/streamlink/blob/master/src/streamlink/plugins/giniko.py

**Reply by ARAB-IPTV** [reply]
> @dct-infra https://github.com/LaneSh4d0w/streamlink/blob/master/src/streamlink/plugins/giniko.py

sorry but i dont understand what you mean by the plugin , i need an m3u8 link , im not prffesional juste amteur

**Reply by BellezaEmporium** [reply]
> > @dct-infra https://github.com/LaneSh4d0w/streamlink/blob/master/src/streamlink/plugins/giniko.py
> 
> sorry but i dont understand what you mean by the plugin , i need an m3u8 link , im not prffesional juste amteur

This is directed to the maintainer of free.fullspeed.tv. I made a plugin that gets the link directly so that once we use free.fullspeed.tv you only need to put the watch link.

**Reply by ARAB-IPTV** [reply]
> > > @dct-infra https://github.com/LaneSh4d0w/streamlink/blob/master/src/streamlink/plugins/giniko.py
> > 
> > 
> > sorry but i dont understand what you mean by the plugin , i need an m3u8 link , im not prffesional juste amteur
> 
> This is directed to the maintainer of free.fullspeed.tv. I made a plugin that gets the link directly so that once we use free.fullspeed.tv you only need to put the watch link.

http://free.fullspeed.tv/iptv-query?streaming-ip=http://www.giniko.com/watch.php?id=328 
dont work

**Reply by BellezaEmporium** [reply]
It's because it hasn't been pushed yet.

**Reply by ARAB-IPTV** [reply]
> It's because it hasn't been pushed yet.

ah okey thank you ismael

**Comment by dct-infra** [comment]
@LaneSh4d0w giniko.py seems to be broken
```
File "/usr/local/lib/python3.9/dist-packages/streamlink/plugins/giniko.py", line 7
    <!DOCTYPE html>
    ^
```

**Reply by BellezaEmporium** [reply]
Hmm... Let me check.

**Reply by BellezaEmporium** [reply]
No issue on my side, could you please give me the link in which you've tested it on ?

Line 6/8 on my side for this link is this : from streamlink.stream.hls import HLSStream

**Reply by dct-infra** [reply]
No specific link, all requests/urls. It blew up QS completely.
We can make a new test in the next few days. I'll let you know.

**Reply by BellezaEmporium** [reply]
> No specific link, all requests/urls. It blew up QS completely.
> We can make a new test in the next few days. I'll let you know.


It works on my local streamlink, though I've made a mistake, it mistook it as a variable stream instead of one only.

**Reply by BellezaEmporium** [reply]
> No specific link, all requests/urls. It blew up QS completely. We can make a new test in the next few days. I'll let you know.

I've updated it. Normally it should work.

**Comment by BellezaEmporium** [comment]
I have a few Algerian channels thanks to my ISP, posting pictures of them to make sure I upload the correct ones.

Algérie 5 (Arabic / Algerian dialect)
![vlcsnap-2022-03-27-22h14m48s486](https://user-images.githubusercontent.com/30985701/160299130-37e06238-18f6-463d-bc73-55722163e805.png)
Algérie 3 (Arabic / Algerian dialect)
![vlcsnap-2022-03-27-22h15m52s013](https://user-images.githubusercontent.com/30985701/160299151-73a32da9-39ba-4733-b95a-243c473fa950.png)
Canal Algérie (Arabic / Algerian dialect & French) 
![vlcsnap-2022-03-27-22h16m24s417](https://user-images.githubusercontent.com/30985701/160299168-7eb40f92-c4a9-4920-b467-f2ed745aeef0.png)

**Reply by ARAB-IPTV** [reply]
Canal Algérie The Second Channel In abic and french 
Algérie 3 " News " 
Algérie 5 " Holly Quran "

**Reply by BellezaEmporium** [reply]
Don't ask me why or how, but the HTA channels came back.

**Comment by BellezaEmporium** [comment]
I've verified, the token is IP-locked in Giniko. I cannot add Canal Algérie, however other links do work (gintna links).

**Reply by ARAB-IPTV** [reply]
can you give me the other links that works with you

**Reply by BellezaEmporium** [reply]
Other links than Canal Algérie, basically all the channels having gintna100.tulix.tv as a host.
